jinahub/indexers/storage/PostgreSQLStorage/postgreshandler.py

```diff
@@ -260,23 +329,110 @@ def get_size(self):
         records = cursor.fetchall()
         return records[0][0]
 
+    def _get_next_shard(self):
```

Comment:
What's the advantage? I've tested this approach pretty thoroughly and it works.

Reply:
Ok, maybe we're using the term incorrectly, but that is what we needed.

Reply:
The advantage is, that different indexers would come to the same conclusion to which shard to write to. It is deterministic with only the Document id and number of total shards as input.

Reply:
So the advantage in our case would be the removal of the extra column, `shard`?

Reply:
Is this hashing function used by mongo available as a package, or do we need to re-implement?

Reply:
Not sure, whether I would remove this column. I think the column should still be there, for a faster dump-generation. Just the calculation should do some hashing instead of round-robin.

Reply:
we can simple employ `hashlib` to implement 

```
import hashlib
def hash_fn(key, total_shards):
    sha = hashlib.sha256()
    sha.update(bytes(key, 'utf-8'))
    return int(sha.hexdigest(), 16) % total_shards
```
this is quite simple, but it works

Reply:
But won't this affect speed?

Reply:
I am not sure, about which point in time we are talking about here. I mean, that you do this hashing during a `/index` call. So only once, when storing the Document the very first time to PSQL. So a drop-in replacement of your the current call to `_get_next_shard`.

Reply:
done, adapted